keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,740 @@
+## COSI Driver gRPC API
 
-If the principal is set, then it should be used as the username of the created credentials or in some way should be deterministically used to generate a new credential for this request. This principal will be used as the unique identifier for deleting this access by calling ProvisionerRevokeBucketAccess
+#### ProvisionerGetInfo
 
-If the `principal` is empty, then a new service account should be created in the backend that satisfies the requested `access_policy`. The username/principal for this service account should be set in the `principal` field of `ProvisionerGrantBucketAccessResponse`.
+This gRPC call responds with the name of the provisioner. The name is used to identify the appropriate driver for a given BucketRequest or BucketAccessRequest.
 
 ```
-message ProvisionerGrantBucketAccessRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
-    
-    map<string,string> bucket_context = 2;  
+    ProvisionerGetInfo
+    |------------------------------------------|       |---------------------------------------|
+    | grpc ProvisionerGetInfoRequest{}         | ===>  | ProvisionerGetInfoResponse{           | 
+    |------------------------------------------|       |   "name": "s3.amazonaws.com"          |
+                                                       | }                                     |
+                                                       |---------------------------------------|
+```
 
-    string principal = 3;
-    
-    // This field is REQUIRED
-    string access_policy = 4;
-}
+#### ProvisionerCreateBucket
 
-message ProvisionerGrantBucketAccessResponse {
-    // This is the account that is being provided access. This will
-    // be required later to revoke access. 
-    string principal = 1;
-    
-    string credentials_file_contents = 2;
-    
-    string credentials_file_path = 3;
-} 
+This gRPC call creates a bucket in the OSP, and returns information about the new bucket. This api must be idempotent. The input to this call is the name of the bucket and an opaque parameters field. 
+
+The returned `bucketID` should be a unique identifier for the bucket in the OSP. This value could be the name of the bucket too. This value will be used by COSI to make all subsequent calls related to this bucket.
+ 
+```
+    ProvisionerCreateBucket
+    |------------------------------------------|       |-----------------------------------------------|
+    | grpc ProvisionerCreateBucketRequest{     | ===>  | ProvisionerCreateBucketResponse{              |
+    |     "name": "br-$uuid",                  |       |   "bucketID": "br-$uuid",                     |
+    |     "parameters": {                      |       |   "bucketInfo": {                             |
+    |        "key": "value"                    |       |      "s3": {                                  |
+    |     }                                    |       |        "bucketName": "br-$uuid",              |
```

Comment:
In the Response, do we need to have both "bucketID" and "bucketName"?  Is "bucketName" always the same as "name" in the Request?

Reply:
They are not always the same. In case of buckets created outside of cosi, they are can be different. In case of COSI created buckets, they are the same